napari/_qt/qt_viewer.py

```diff
@@ -1059,7 +1069,69 @@ def dropEvent(self, event):
             else:
                 filenames.append(url.toString())
 
-        self.viewer.open(filenames, stack=bool(shift_down))
+        # if trying to open as a stack, open with any available reader
+        if shift_down:
+            self.viewer.open(filenames, stack=bool(shift_down))
+            return
+
+        for filename in filenames:
+            # get available readers for this file from all registered plugins
+            readers = get_potential_readers(filename)
+            if not readers:
+                raise ValueError(
+                    trans._(
+                        'No plugin found capable of reading {filename}.',
+                        deferred=True,
+                        filename=filename,
+                    )
+                )
+
```

Comment:
i see that `if len(readers) == 1` is checked later ...  curious, why not just do it here?

Reply:
I guess I'm doing it later because I wanted to give preference to whatever settings the user may have saved. Further down, if there is just one reader but we had an existing setting that failed, we still let the user know.